packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/TypeProvider.cs

```diff
@@ -175,5 +182,10 @@ protected virtual XmlDocProvider BuildXmlDocs()
                 _constructors = (constructors as IReadOnlyList<ConstructorProvider>) ?? constructors.ToList();
             }
         }
+        public IReadOnlyList<EnumTypeMember> EnumValues => _enumValues ??= BuildEnumValues();
+
+        protected virtual IReadOnlyList<EnumTypeMember> BuildEnumValues() => throw new InvalidOperationException("Not an EnumProvider type");
+
+        private IReadOnlyList<EnumTypeMember>? _enumValues;
```

Comment:
`TypeProvider` could be an enum, but it is not an enum.
Therefore I think these properties do not make sense here.
Only enums should define these properties.

I wonder why the `CSharpType` is expecting we could pass in a `EnumUnderlyingType`. Where is it used? It does not look quite right.

Reply:
This is similar to Stream CanSeek and Seek.  We are providing a bool check IsEnum that a user should call to know whether or not its safe to call EnumValues.